grails-app/assets/javascripts/streama/controllers/settings-settings-ctrl.js

```diff
@@ -24,29 +24,32 @@ angular.module('streama')
             window.location.reload();
             alertify.success('Settings saved.');
           })
-      };
-
+        };
 
-      $scope.validateSettings = function (settings) {
-        if($scope.loading === true){
-          return
-        }
 
-        $scope.changeValue(settings);
-        $scope.loading = true;
+        $scope.validateSettings = function (settings) {
+          if ($scope.loading === true) {
+            return
+          }
+          $scope.changeValue(settings);
+          $scope.loading = true;
 
-        apiService.settings.validateSettings(settings).then(function (response) {
+          apiService.settings.validateSettings(settings).then(function (response) {
             var data = response.data;
-            alertify.success(data.message || 'validation successful');
+            if (data.message.includes("Invalid credentials") || data.message.includes("problems")) {
+              alertify.log(data.message)
```

Comment:
seems to me like youre handling error-messages in the success callback - is this done the same way elsewhere? seems like this should have a differetnt statuscode